src/main/webapp/app/exercises/programming/manage/programming-exercise-edit-selected.component.ts

```diff
@@ -0,0 +1,113 @@
+import { Component, OnInit } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { TranslateService } from '@ngx-translate/core';
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
+import { ProgrammingExerciseSimulationService } from 'app/exercises/programming/manage/services/programming-exercise-simulation.service';
+import { ProgrammingExerciseService } from 'app/exercises/programming/manage/services/programming-exercise.service';
+import { JhiAlertService } from 'ng-jhipster';
+
+@Component({
+    selector: 'jhi-programming-exercise-edit-selected',
+    templateUrl: './programming-exercise-edit-selected.component.html',
+})
+export class ProgrammingExerciseEditSelectedComponent implements OnInit {
+    newProgrammingExercise: ProgrammingExercise;
+    selectedProgrammingExercises: ProgrammingExercise[];
+
+    isSaving = false;
+    savedExercises = 0;
+    failedExercises: string[] = [];
+    private translationBasePath = 'artemisApp.programmingExercise.';
+    notificationText: string | null;
```

Comment:
In TypeScript we prefer to use `undefined` instead of `null`
```suggestion
    notificationText: string | undefined;
```